libnetwork/portallocator/portallocator.go

```diff
@@ -114,7 +115,7 @@ func (p *PortAllocator) RequestPort(ip net.IP, proto string, port int) (int, err
 // default IP (0.0.0.0).
 func (p *PortAllocator) RequestPortInRange(ip net.IP, proto string, portStart, portEnd int) (int, error) {
```

Comment:
Excluding tests, `RequestPortInRange` is only called by `RequestPort`. `RequestPort` is only called from `allocateDaemonPort` in a loop that already prevents `ip` to be `nil`.

I think we can just drop that defaulting behavior from this method, but it's probably worth taking a defensive stance by returning an error if `ip` is nil.

Reply:
I think I had a patch stashed that removed that loop, which I left out of this PR for now, but, yes, also agreed that we could look at reducing the signature, and remove what we don't need.